src/test/scala/nodes/learning/LeastSquaresEstimatorSuite.scala

```diff
@@ -0,0 +1,92 @@
+package nodes.learning
+
+import breeze.linalg.{DenseVector, SparseVector}
+import org.apache.spark.SparkContext
+import org.scalatest.FunSuite
+import pipelines._
+import workflow.WorkflowUtils
+
+class LeastSquaresEstimatorSuite extends FunSuite with LocalSparkContext with Logging {
```

Comment:
Very nice unit tests ! Minor question: 

1. If you set `k = 2` as in binary classification does DenseLBFGS get chosen over the BlockSolver ?
2. Does this test take very long to run ? Just curious if it'll be fine to put in Jenkins etc.